chainntnfs/neutrinonotify/neutrino.go

```diff
@@ -318,87 +319,40 @@ func (n *NeutrinoNotifier) notificationDispatcher() {
 				n.blockEpochClients[msg.epochID] = msg
 			}
 
-		case item := <-n.newBlocks.ChanOut():
-			newBlock := item.(*filteredBlock)
-
-			n.heightMtx.Lock()
-			n.bestHeight = newBlock.height
-			n.heightMtx.Unlock()
-
-			chainntnfs.Log.Infof("New block: height=%v, sha=%v",
-				newBlock.height, newBlock.hash)
-
-			// First we'll notify any subscribed clients of the
-			// block.
-			n.notifyBlockEpochs(int32(newBlock.height), &newBlock.hash)
-
-			// Next, we'll scan over the list of relevant
-			// transactions and possibly dispatch notifications for
-			// confirmations and spends.
-			for _, tx := range newBlock.txns {
-				// Check if the inclusion of this transaction
-				// within a block by itself triggers a block
-				// confirmation threshold, if so send a
-				// notification. Otherwise, place the
-				// notification on a heap to be triggered in
-				// the future once additional confirmations are
-				// attained.
-				mtx := tx.MsgTx()
-				txIndex := tx.Index()
-				txSha := mtx.TxHash()
-				n.checkConfirmationTrigger(&txSha, newBlock, txIndex)
-
-				for i, txIn := range mtx.TxIn {
-					prevOut := txIn.PreviousOutPoint
-
-					// If this transaction indeed does
-					// spend an output which we have a
-					// registered notification for, then
-					// create a spend summary, finally
-					// sending off the details to the
-					// notification subscriber.
-					if clients, ok := n.spendNotifications[prevOut]; ok {
-						// TODO(roasbeef): many
-						// integration tests expect
-						// spend to be notified within
-						// the mempool.
-						spendDetails := &chainntnfs.SpendDetail{
-							SpentOutPoint:     &prevOut,
-							SpenderTxHash:     &txSha,
-							SpendingTx:        mtx,
-							SpenderInputIndex: uint32(i),
-							SpendingHeight:    int32(newBlock.height),
-						}
-
-						for _, ntfn := range clients {
-							chainntnfs.Log.Infof("Dispatching "+
-								"spend notification for "+
-								"outpoint=%v", ntfn.targetOutpoint)
-							ntfn.spendChan <- spendDetails
+		case item := <-n.chainUpdates.ChanOut():
+			update := item.(*filteredBlock)
+			if update.connect {
+				n.heightMtx.Lock()
+				if update.height != n.bestHeight+1 {
+					chainntnfs.Log.Warnf("Received blocks out of order: "+
+						"current height=%d, new height=%d",
+						n.bestHeight, update.height)
+				}
 
-							// Close spendChan to ensure that any calls to Cancel will not
-							// block. This is safe to do since the channel is buffered, and the
-							// message can still be read by the receiver.
-							close(ntfn.spendChan)
-						}
+				n.bestHeight = update.height
+				n.heightMtx.Unlock()
 
-						delete(n.spendNotifications, prevOut)
-					}
+				chainntnfs.Log.Infof("New block: height=%v, sha=%v",
+					update.height, update.hash)
 
+				err := n.handleBlockConnected(update)
+				if err != nil {
+					chainntnfs.Log.Error(err)
+				}
```

Comment:
Similar comment here about avoiding the nesting by using a `continue` after the first clause.

Reply:
Yes, technically true, but I don't think it makes sense in this context. The first block handles the case of a block connection, the second handles a block disconnection. The logic is symmetrical (like it would make sense if the if was `!update.connect` and the blocks were flipped), so I'd rather have them at the same level of indentation.

Reply:
Removing the nesting will reduce the number of lines in the diff. It's not a block as it's mostly a style thing, but in the codebase we generally try to limit the degree of nesting in a given function.